main.go

```diff
@@ -61,15 +64,21 @@ func main() {
 		return
 	}
 
-	viper.SetConfigName("config") // name of config file (without extension)
-	viper.SetConfigType("yaml")   // REQUIRED if the config file does not have the extension in the name
-	viper.AddConfigPath(".")      // optionally look for config in the working directory
-	err := viper.ReadInConfig()   // Find and read the config file
-	if err != nil {               // Handle errors reading the config file
-		fmt.Printf("failed to read config file config.yaml: %v \n", err)
-		return
+	if regtest {
```

Comment:
Then you could do here this

`
	configName := "config"
	if config != nil {
		configName = *config
	}
	viper.SetConfigName(configName) // name of config file (without extension)
`